lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -1818,8 +1818,31 @@ bool TypeSystemSwiftTypeRef::IsImportedType(opaque_compiler_type_t type,
   VALIDATE_AND_RETURN(impl, IsImportedType, type,
                       (ReconstructType(type), nullptr));
 }
+
 CompilerType TypeSystemSwiftTypeRef::GetErrorType() {
-  return m_swift_ast_context->GetErrorType();
+  auto impl = [&]() {
+    using namespace swift::Demangle;
+    Demangler Dem;
+    llvm::ArrayRef<NodePointer> nodes = {
+        Dem.createNode(Node::Kind::Type),
+        Dem.createNode(Node::Kind::ProtocolList),
+        Dem.createNode(Node::Kind::TypeList),
+        Dem.createNode(Node::Kind::Type),
+        Dem.createNode(Node::Kind::Protocol),
+    };
+    for (int i = 1; i < nodes.size(); ++i)
+      nodes[i - 1]->addChild(nodes[i], Dem);
```

Comment:
That's a nice idea for creating a chain of nodes. You might also want to check out the helper functions used by the unit tests (TestTypeSystemSwiftTypeRef.cpp) and see if that is cleaner/shorter/or worse.

Reply:
The `Node()` constructor helper function looks nice. Are you suggesting reusing that function here? In fact there's also an expression that constructs `Error` in the test, so that could be reused too.

Reply:
I got rid of the looped chain of nodes to not block this on deciding on a construction abstraction in this file. I think it can be a follow up, there are other patterns in this file.